### PR TITLE
Add get_stake_minimum_delegation() to rpc_client

### LINKED ISSUE #24690: Add a way to get stake minimum delegation from RPC
### Problem

From RPC, we cannot query the minimum stake delegation.

#### Proposed Solution

Add a new RPC endpoint.

We cannot use `simulate_transaction()` here, as it fails when loading accounts.

#### Previous proposed solution

> How about an `RpcClient` helper under the hood that performs `simulate_transaction` and returns parsed `return_data` of type T? I think we're going to want something similar for `spl_token::instruction::GetAccountDataSize`

_Originally posted by @CriesofCarrots in https://github.com/solana-labs/solana/issues/24690#issuecomment-1109933456_

#### Initial Description

> how about moving this into solana-program, so we don’t need a new dep in ledger/
When it’s moved into solana-program, we could probably remove the FeatureSet as well (since that’s only available in solana-sdk) and instead just always use 1 SOL

_Originally posted by @mvines in https://github.com/solana-labs/solana/pull/24660#discussion_r858083345_

### PR DESCRIPTION
**Pull request by brooksprumo**
#### Problem

There is not a way for rpc client to get the stake minimum delegation.

#### Summary of Changes

Add `get_stake_minimum_delegation()` to rpc_client.

Fixes #24690

### CODE REVIEW

File: client/src/nonblocking/rpc_client.rs [hunk 1]
```diff
@@ -4507,6 +4508,42 @@ impl RpcClient {
...
+    pub async fn get_stake_minimum_delegation(&self) -> ClientResult<u64> {
+        let instruction = solana_sdk::stake::instruction::get_minimum_delegation();
+        let payer = None;
+        let transaction = Transaction::new_with_payer(&[instruction], payer);
+        let response = self.simulate_transaction(&transaction).await?;
+        let return_data = response
+            .value
+            .return_data
+            .ok_or_else(|| ClientErrorKind::Custom("return data was empty".to_string()))?;
+        let (program_id, data) = (return_data.program_id, return_data.data);
```
Comment (CriesofCarrots) [inline client/src/nonblocking/rpc_client.rs:4521]:
```suggestion
        let (program_id, (data, encoding)) = response
            .value
            .return_data
            .ok_or_else(|| ClientErrorKind::Custom("return data was empty".to_string()))?;
```

Comment (CriesofCarrots) [reply client/src/nonblocking/rpc_client.rs:4521]: Wdyt? Slightly less verbose, and avoids the need for tuple indexing on `data` below.

**Review by CriesofCarrots — APPROVED** [review]
Looks good! Just nits

File: client/src/nonblocking/rpc_client.rs [hunk 2]
```diff
@@ -4507,6 +4508,42 @@ impl RpcClient {
         parse_keyed_accounts(accounts, RpcRequest::GetProgramAccounts)
     }
 
+    /// Returns the stake minimum delegation, in lamports.
+    pub async fn get_stake_minimum_delegation(&self) -> ClientResult<u64> {
+        let instruction = solana_sdk::stake::instruction::get_minimum_delegation();
+        let payer = None;
```
[the remark this answers is not in the document]
Comment (brooksprumo) [reply client/src/nonblocking/rpc_client.rs:4514]: Nope. Removed.

File: client/src/nonblocking/rpc_client.rs (hunk 1 above)
Comment (brooksprumo) [reply client/src/nonblocking/rpc_client.rs:4521]: Love it! Done.

**Pull request merged**